docs/resources/equinix_network_file.md

```diff
@@ -0,0 +1,54 @@
+---
+subcategory: "Network Edge"
+---
+
+# equinix_network_file (Resource)
+
+Resource `equinix_network_file` allows creation and management of Equinix Network Edge files.
+
+## Example Usage
+
+```hcl
+variable "filepath" { default = "fileFolder/fileName.txt" }
+
+resource "equinix_network_file" "test-file" {
+  file_name = "fileName.txt"
+  content = file("${path.module}/${var.filepath}")
+  metro_code = "SV"
+  device_type_code = "AVIATRIX_EDGE"
+  process_type = "CLOUD_INIT"
+  self_managed = true
+  byol = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `file_name` - (Required) File name.
+* `content` - (Required) Uploaded file content, expected to be a UTF-8 encoded string.
+* `metro_code` - (Required) File upload location metro code. It should match the device location metro code.
+* `type_code` - (Required) Device type code.
+* `process_type` - (Required) File process type (LICENSE or CLOUD_INIT).
+* `self_managed` - (Optional) Boolean value that determines device management mode, i.e.,
```

Comment:
I think the default is `false` for `self_managed`.  I'd recommend specifying the mapping (`true` for self-managed, `false` for Equinix managed).

Reply:
Mapping is already defined in the code. Docs here mention that it's a boolean value. As discussed, to avoid confusion with the device resource, both self_managed and byol has been marked as required fields. No default will be provided. They are all boolean value similar to what we have in device resource.